pipes/rules/assembly.rules

```diff
@@ -30,67 +30,116 @@ rule all_assemble_failures:
     params: LSF="-N"
 
 
+rule trim_rmdup_subsamp:
+    ''' Trim reads with trimmomatic, rmdup with prinseq,
+        and random subsample to no more than 100k reads.
+    '''
+    input:  config["data_dir"]+'/'+config["subdirs"]["per_sample"]+'/{sample}.taxfilt.bam'
+    output: subsamp_bam=config["tmp_dir"] +'/'+config["subdirs"]["assembly"]+'/{sample}.subsamp.bam'
+    resources: 
+            mem=7,
+            cores=int(config.get("number_of_threads", 1))
+    params: LSF=config.get('LSF_queues', {}).get('short', '-W 4:00'),
+            UGER=config.get('UGER_queues', {}).get('short', '-q short'),
+            n_reads=str(config["trinity_n_reads"]),
+            logid="{sample}",
+            clipDb=config["trim_clip_db"]
+    run:
+            shell("{config[bin_dir]}/assembly.py trim_rmdup_subsamp {input} {params.clipDb} {output.subsamp_bam}  --outStatsFile={output.subsamp_bam}.stats --n_reads={params.n_reads}")
 
 rule assemble_trinity:
     ''' This step runs the Trinity assembler.
         First trim reads with trimmomatic, rmdup with prinseq,
         and random subsample to no more than 100k reads.
     '''
-    input:  config["data_dir"]+'/'+config["subdirs"]["per_sample"]+'/{sample}.taxfilt.bam'
+    input:  config["tmp_dir"]+'/'+config["subdirs"]["assembly"]+'/{sample}.'+config.get("assembly_preproc_method","subsamp")+'.bam'
```

Comment:
Can do that.   For the pipeline, I'd keep preprocessing as a separate step, since e.g. multiple assemblers might use  the preprocessed  reads.  I also like the Unix way of having well-defined individual commands that can  be chained.   But if people are  also routinely running individual CLI tools, then aggregate commands like preprocess-and-assemble make sense.